### PR TITLE
Feat/test job offers filters

### PR DESCRIPTION
**Pull request by KapiSolutions**

### CODE REVIEW

File: tests/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+// Check if offer list is filtered according to setted filter
+const isOfferListFiltered = async (filter: FilterData): Promise<boolean> => {
+   // Get currently displayed offer list
+   const offers = await getOffersData();
+   // Loop over offers and check if there are filtered offers
+   if (offers.length > 0) {
+      for (const offer of offers) {
+         if (typeof offer[filter.offerKeyName] === 'number' && typeof filter.value === 'number') {
+            // Compare number values
+            if (filter.offerOption.includes('-min')) {
```
Comment (Sumick) [inline tests/utils/helpers.ts:126]: **nitpick:** This is very complicated, and unreadable - I would split it explicitly for each filter, so it's clear what's going on. Eg. `isJobTypeSelected`

File: tests/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+               }
+            }
+         } else if (
+            typeof offer[filter.offerKeyName] === 'string' &&
+            typeof filter.value === 'boolean'
+         ) {
+            // When checkbox is checked then compare offer data with setted to filter
+            //  (eg. When "Job Type" filter checkbox eg. "Contract" is checked then
+            //      offer.jobType should be equal to Contract, and only that offers should be displayed )
+            if (
```
Comment (Sumick) [inline tests/utils/helpers.ts:146]: **issue:** This testing function would need unit testing in the current state :D

File: tests/utils/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            typeof filter.value === 'string'
+         ) {
+            // Check input inputs agains proper filtering.
+            // Offerlist should display only that offers which includes provided input text(job title/location)
+            if (
+               !(offer[filter.offerKeyName] as string)
+                  .toLowerCase()
+                  .includes((filter.value as string).toLowerCase())
+            ) {
+               // Values doesn't match, offer list is incorrectly filtered
```
Comment (Sumick) [inline tests/utils/helpers.ts:165]: **nitpick:** This monster needs to be killed :D

File: tests/utils/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+         // Each [number] filter element should set to 0
+         if (typeof dataBefore.value === 'number' && dataBefore.value !== 0) {
+            return false;
+         }
+      }
+   }
+   return true;
+};
+
+export const setRandomFilters = async () => {
```
Comment (Sumick) [inline tests/utils/helpers.ts:199]: **issue:** What filters? It's impossible to track, and maintain. Let's split it to `setRandomLocationFilter`, `setRandomJobTypeFilter`, etc.

**Review by Sumick — CHANGES_REQUESTED** [review]
We need to simplify these tests, improve readability, and maintainability. Currently these tests are too generic. You made helper functions that iterates over elements dynamically - it's not a good idea because you don't have control over it. For sure, we need to split these dynamic checks into small, understandable pieces.

File: tests/utils/helpers.ts (hunk 4 above)
Comment (KapiSolutions) [reply tests/utils/helpers.ts:199]: All filters under "filters-section". So all filters as we have only one filters section in app.
`
export const setRandomFilters = async () => {
   const filters = screen.queryAllByRole('filters-section');
   for (const filter of filters) {...}}
`

It's possible to track by adding console.log with currently failed testing component. And don't need to change anything here in order to extending filters in app (eg. new country filters, this function will automatically test them too). 

This was my idea, to be able to test new filters immediately after adding them to the application - without having to write new tests for each new type.

@Sumick, bearing the above in mind - should I leave the current solution or make the suggested changes?

File: tests/utils/helpers.ts (hunk 3 above)
Comment (KapiSolutions) [reply tests/utils/helpers.ts:165]: Haha yes 😂 that's how to become hero - creating monsters and killing them😂

**Review by Sumick — APPROVED** [review]
Much, much better! Amazing job 🔥🙌

**Pull request merged**